### PR TITLE
Document how to create and publish new plugin

### LINKED ISSUE #792: Document how to create and publish a new plugin
### What should we change and why?

Building on #785 and #791 , update the documentation template to include instructions on how users can create and publish new Arelle plugins to PyPI using the ixbrl-viewer as an example.

The documentation should include the [__pluginInfo__](https://github.com/Arelle/Arelle/blob/ea2c2b80243189bc0855895618f6913625cd3234/arelle/plugin/validate/ESEF_2022/__init__.py#L1301-L1321) signature along with all of the [pluginClassMethods](https://github.com/Arelle/Arelle/blob/ea2c2b80243189bc0855895618f6913625cd3234/arelle/PluginManager.py#L472) hooks available for plugins.

### PR DESCRIPTION
**Pull request by aaroncameron-wk**
#### Reason for change
Closes #792 

#### Description of change
Also added documentation on _using_ plugins as it's pertinent to plugin development.
Added initial plugin hooks documentation but did not document all hooks, as I think that's better left to others who have more specific knowledge of those plugin areas.

#### Steps to Test
Generate docs locally and view new documentation under `Plugins`

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: docs/source/plugins/installation.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Plugin Installation
+
+:::{index} Plugin Installation
+:::
+
+## Preinstalled Plugins
+Some plugins are built into and packaged with Arelle.
+These plugins are located in source under `/arelle/plugins`.
```
Comment (austinmatherne-wk) [inline docs/source/plugins/installation.md:8]: A link to the directory on GitHub would be good.

File: docs/source/plugins/installation.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+No additional installation steps are necessary for these plugins.
+Activate pre-installed plugins [via the GUI using Select][gui-select] or [via CLI by name][cli-name].
+
+## Installing Plugins via Package Manager
+Published plugins can be installed by your Python package manager of choice.
```
Comment (austinmatherne-wk) [inline docs/source/plugins/installation.md:13]: Should probably mention that this only works if you're running Arelle from source with your own Python environment (project:install.md#from-python-source).

File: docs/source/plugins/installation.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+```
+Activate plugins installed with this method [via the GUI using Select][gui-select] or [via CLI by name][cli-name].
+
+## Installing Plugins Manually
+If the plugin you are using is not available via package manager, 
+or you would like to run a plugin you are developing locally,
+plugins can also be installed anywhere Arelle can access on your local file system.
+You can then activate manually installed plugins [via the GUI using Browse][gui-browse] or [via CLI by path][cli-path].
+
+Alternatively, you can place your plugin in the preinstalled plugins directory (`/arelle/plugins`)
```
Comment (austinmatherne-wk) [inline docs/source/plugins/installation.md:27]: nit: use a relative path in the example `arelle/plugins`

File: docs/source/plugins/using_with_cli.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+## Enabling Plugins for the CLI
+Plugin enablement does not persist between CLI runs, 
+so every CLI execution needs any used plugins to be specified each time.
+
+[//]: # (TODO: link below --plugins mentions to CLI documentation once it's implemented)
+
+___
+### By Name
+The plugin's name can be passed to the `--plugins` CLI option.
```
Comment (austinmatherne-wk) [inline docs/source/plugins/using_with_cli.md:14]: Should also probably mention that multiple plugins can be specified by providing them as a `|` separated string. `python arelleCmdLine.py --plugins "myplugin1|myplugin2"  ...`

File: docs/source/plugins/development/publishing.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+Use the [Python Packaging User Guide](https://packaging.python.org/en/latest/) to learn how to package and publish a package to PyPI.
+Specifically, the [Packaging Python Projects](https://packaging.python.org/en/latest/tutorials/packaging-projects/) details this process.
+
+## Configuring Entry Point
+Publishing the package will make it installable by package managers, 
+but an extra step is necessary to make the package discoverable as a plugin by Arelle.
+
+Arelle uses [`setuptools`](https://pypi.org/project/setuptools/) to discover packages that define entry points for Arelle.
+Entry points can be defined in `pyproject.toml` as described [here](https://setuptools.pypa.io/en/latest/userguide/entry_point.html#entry-points-for-plugins).
+For `ixbrl-viewer`, that looks like [this](https://github.com/Arelle/ixbrl-viewer/blob/master/pyproject.toml#L41-L43C1):
```
Comment (austinmatherne-wk) [inline docs/source/plugins/development/publishing.md:18]: Use a permalink URL. The referenced lines will end up out of sync at some point.

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**